ml/dgl_utils.py

```diff
@@ -0,0 +1,253 @@
+# classes for converting a dataframe or Graphistry Plottable into a DGL
+import pandas as pd
+import dgl
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import umap
+import numpy as np
+
+from ml.utils import pandas2dgl
+from ml.utils import process_dirty_dataframes
+from ml import constants as config
+
+import logging
+
+logger = logging.getLogger(__name__)
+FORMAT = "[%(filename)s:%(lineno)s - %(funcName)20s() ] %(message)s"
+logging.basicConfig(format=FORMAT)
+logger.setLevel(logging.DEBUG)
+
+
+# ndata_fields = ['feature', 'label', 'train_mask', 'test_mask', 'validation_mask']
+# edata_fields = ['feature', 'relationship', 'train_mask', 'test_mask', 'validation_mask']
+
+
+def get_vectorizer(name):
+    if type(name) != str:
+        logger.info(f"Returning {type(name)} vectorizer")
+        return name
+    if name == config.DIRTY_CAT:
+        return process_dirty_dataframes
+    if name == config.SKLEARN:
+        # TODO
+        return
+    logging.warning(
+        f"Vectorizer name must be one of [{config.DIRTY_CAT}, {config.SKLEARN}, ..]"
+    )
+    return
+
+
+def get_dataframe_for_target(target, df, name):
+    if type(target) == str:
+        logger.info(f"Returning {name}-target {target} as DataFrame")
+        return pd.DataFrame({target: df[target].values}, index=df.index)
+    logger.info(f"Returning {name}-target as itself")
+    return target
+
+
+def check_target_not_in_features(df, y, remove=True):
+    if y is None:
+        return df, y
+    remove_cols = []
+    if hasattr(y, "columns"):
+        yc = y.columns
+        xc = df.columns
+        for c in yc:
+            if c in xc:
+                remove_cols.append(c)
+    else:
+        logger.warning(f"Target is not of type(DataFrame) and has no columns")
+    if remove:
+        logger.info(f"Removing {remove_cols} columns from DataFrame")
+        tf = df.drop(columns=remove_cols)
+        return tf, y
+    return df, y
+
+
+def featurization(df, y, vectorizer, remove=True):
+    df, y = check_target_not_in_features(df, y, remove=remove)
+    X_enc, y_enc, sup_vec, sup_label = vectorizer(df, y)
+    if y_enc is not None:
+        y_enc = torch.tensor(y_enc.values)
+        data = {config.FEATURE: torch.tensor(X_enc.values), config.TARGET: y_enc}
+    else:
+        data = {config.FEATURE: torch.tensor(X_enc.values)}
+    encoders = {config.FEATURE: sup_vec, config.TARGET: sup_label}
+    return data, encoders
+
+
+def scatterplot(ux, color_labels=None):
```

Comment:
let's drop any non-graphistry viz from pygraphistry: people can take the df's and plot with something else if they don't want pygraphistry's

for the case of a scatterplot, can use a graphistry plot with node x/y positions and play=0, see demos folders